### ISSUE TITLE
Duplicated reserved GTF columns with certain attribute tags in `write_gtf`

### ISSUE THREAD

**Issue by fairliereese** [pyranges/pyranges#341]
Hi, 

I'm having issues when writing out a GTF that has an attribute that is named in `_gtf_columns` already. 

GTF snippet:
```
chr1	HAVANA	transcript	3206523	3215632	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_name "Xkr4"; gene_status "KNOWN"; gene_type "protein_coding"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; talon_gene "3"; talon_transcript "4"; havana_transcript "OTTMUST00000086624.1"; level "2"; source "HAVANA"; transcript_support_level "1";
chr1	HAVANA	exon	3213439	3215632	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "Xkr4"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; exon_number "1"; exon_id "ENSMUSE00000863980.1"; talon_gene "3"; talon_transcript "4"; talon_exon "6"; exon_status "KNOWN"; level "2"; source "HAVANA";
chr1	HAVANA	exon	3206523	3207317	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "Xkr4"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; exon_number "2"; exon_id "ENSMUSE00000867897.1"; talon_gene "3"; talon_transcript "4"; talon_exon "8"; exon_status "KNOWN"; level "2"; source "HAVANA";
chr1	HAVANA	transcript	3375556	3377788	.	-	.	gene_id "ENSMUSG00000104017.1"; transcript_id "ENSMUST00000192336.1"; gene_name "Gm37363"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37363-201"; talon_gene "6"; talon_transcript "8"; havana_transcript "OTTMUST00000127146.1"; level "2"; source "HAVANA"; tag "basic"; transcript_support_level "NA";
chr1	HAVANA	exon	3375556	3377788	.	-	.	gene_id "ENSMUSG00000104017.1"; transcript_id "ENSMUST00000192336.1"; gene_type "TEC"; gene_status "KNOWN"; gene_name "Gm37363"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37363-201"; exon_number "1"; exon_id "ENSMUSE00001343686.1"; talon_gene "6"; talon_transcript "8"; talon_exon "16"; exon_status "KNOWN"; level "2"; source "HAVANA"; tag "basic";
```

As you can see, one of the attribute tags is "source", which is used in `write_gtf` to try and identify and order the necessary columns in the GTF table before writing to output. The result of this is that the source column ends up duplicating in the saved GTF, as you can see here:
```
chr1	HAVANA	HAVANA	transcript	3206523	3215632	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_name "Xkr4"; gene_status "KNOWN"; gene_type "protein_coding"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; talon_gene "3"; talon_transcript "4"; havana_transcript "OTTMUST00000086624.1"; level "2"; transcript_support_level "1";
chr1	HAVANA	HAVANA	exon	3213439	3215632	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_name "Xkr4"; gene_status "KNOWN"; gene_type "protein_coding"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; talon_gene "3"; talon_transcript "4"; level "2"; exon_number "1"; exon_id "ENSMUSE00000863980.1"; talon_exon "6"; exon_status "KNOWN";
chr1	HAVANA	HAVANA	exon	3206523	3207317	.	-	.	gene_id "ENSMUSG00000051951.5"; transcript_id "ENSMUST00000159265.1"; gene_name "Xkr4"; gene_status "KNOWN"; gene_type "protein_coding"; transcript_type "processed_transcript"; transcript_status "KNOWN"; transcript_name "Xkr4-202"; talon_gene "3"; talon_transcript "4"; level "2"; exon_number "2"; exon_id "ENSMUSE00000867897.1"; talon_exon "8"; exon_status "KNOWN";
chr1	HAVANA	HAVANA	transcript	3375556	3377788	.	-	.	gene_id "ENSMUSG00000104017.1"; transcript_id "ENSMUST00000192336.1"; gene_name "Gm37363"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37363-201"; talon_gene "6"; talon_transcript "8"; havana_transcript "OTTMUST00000127146.1"; level "2"; transcript_support_level "NA"; tag "basic";
chr1	HAVANA	HAVANA	exon	3375556	3377788	.	-	.	gene_id "ENSMUSG00000104017.1"; transcript_id "ENSMUST00000192336.1"; gene_name "Gm37363"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37363-201"; talon_gene "6"; talon_transcript "8"; level "2"; exon_number "1"; exon_id "ENSMUSE00001343686.1"; talon_exon "16"; exon_status "KNOWN"; tag "basic";
chr1	HAVANA	HAVANA	transcript	3464977	3467285	.	-	.	gene_id "ENSMUSG00000103025.1"; transcript_id "ENSMUST00000194099.1"; gene_name "Gm37686"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37686-201"; talon_gene "7"; talon_transcript "9"; havana_transcript "OTTMUST00000127101.1"; level "2"; transcript_support_level "NA"; tag "basic";
chr1	HAVANA	HAVANA	exon	3464977	3467285	.	-	.	gene_id "ENSMUSG00000103025.1"; transcript_id "ENSMUST00000194099.1"; gene_name "Gm37686"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm37686-201"; talon_gene "7"; talon_transcript "9"; level "2"; exon_number "1"; exon_id "ENSMUSE00001337180.1"; talon_exon "17"; exon_status "KNOWN"; tag "basic";
chr1	HAVANA	HAVANA	transcript	3592892	3595903	.	-	.	gene_id "ENSMUSG00000103161.1"; transcript_id "ENSMUST00000195166.1"; gene_name "Gm38148"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm38148-201"; talon_gene "11"; talon_transcript "13"; havana_transcript "OTTMUST00000127098.1"; level "2"; transcript_support_level "NA"; tag "basic";
chr1	HAVANA	HAVANA	exon	3592892	3595903	.	-	.	gene_id "ENSMUSG00000103161.1"; transcript_id "ENSMUST00000195166.1"; gene_name "Gm38148"; gene_status "KNOWN"; gene_type "TEC"; transcript_type "TEC"; transcript_status "KNOWN"; transcript_name "Gm38148-201"; talon_gene "11"; talon_transcript "13"; level "2"; exon_number "1"; exon_id "ENSMUSE00001343966.1"; talon_exon "23"; exon_status "KNOWN"; tag "basic";
chr1	HAVANA	HAVANA	transcript	4492465	4493735	.	-	.	gene_id "ENSMUSG00000025902.13"; transcript_id "ENSMUST00000191939.1"; gene_name "Sox17"; gene_status "KNOWN"; gene_type "protein_coding"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "Sox17-204"; talon_gene "20"; talon_transcript "32"; havana_transcript "OTTMUST00000127266.2"; level "2"; transcript_support_level "1"; tag "cds_end_NF"; protein_id "ENSMUSP00000142154.1";
```

Minimal code: 
```python
import pyranges as pr
df = pr.read_gtf('test_input.gtf', duplicate_attr=True)
df.to_gtf('test_output.gtf')
```

I'm not sure at what level it would be best to address this. I can imagine that this could be done as input sanitization perhaps?

**Comment by endrebak** [comment]
Nice catch.

To begin with I propose warning the user if they are duplicated and let them deal with it. It should not be automatically handled in silence I think.

**Comment by fairliereese** [comment]
OK! I made some changes to my fork (https://github.com/fairliereese/pyranges). I added a `rename_attr` bool option (default False) to `read_gtf` which will

- If false, throw an error informing the user of the reserved column names in the attributes keys and exit
- If true, warn the user of the reserved column names in the attribute, and update said reserved column names with the suffix "_attr" (with a message to the user that it is doing so). 

If this behavior seems like what you're looking for I'm happy to open a PR!
